### PR TITLE
#2-Implemented basic 3-tier architecture and dependencies

### PR DESCRIPTION
**Pull request by xDAndrew**
##### TRELLO INFO

N2 - https://trello.com/c/q6OGBbyW/2-introduce-basic-3-tier-architecture-and-dependencies

##### CHANGE TYPE

- New feature (non-breaking change which adds functionality)

##### DESCRIPTION

- Added projects:
    - ReKreator.MVC - Web Interface for application;
    - ReKreator.BusinessLogic - Contains the application's business logic;
    - ReKreator.DataAccessLayer - Provides access to data;
- Added base references between projects;
- Added NuGet plugins:
    - Bootstrap
    - Autofac.MVC5
    - Autofac.Owin (for Identity)
    - EntityFramework 6.2
    - Identity 2.0
- Removed test project

### CODE REVIEW

**Comment by tonnypeperonny** [comment]
Looks good.

File: src/ReKreator.HtmlParser.Tests/ReKreator.HtmlParser.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    <Compile Include="Properties\AssemblyInfo.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="app.config" />
+    <None Include="packages.config">
+      <SubType>Designer</SubType>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\ReKreator.BusinessLogic\ReKreator.BusinessLogic.csproj">
```
Comment (AlexKonon) [inline src/ReKreator.HtmlParser.Tests/ReKreator.HtmlParser.Tests.csproj]: Hmm, do we really need to reference BLL to test parsing only? 
Please fix

File: src/ReKreator.MailDelivery.Tests/ReKreator.MailDelivery.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  <ItemGroup>
+    <None Include="app.config" />
+    <None Include="packages.config" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\ReCreator.MailDelivery\ReCreator.MailDelivery.csproj">
+      <Project>{b1e9ef5f-a96f-4bd9-a1ea-7889237f4749}</Project>
+      <Name>ReCreator.MailDelivery</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\ReKreator.BusinessLogic\ReKreator.BusinessLogic.csproj">
```
Comment (AlexKonon) [inline src/ReKreator.MailDelivery.Tests/ReKreator.MailDelivery.Tests.csproj]: The same here.

File: src/ReKreator.DataAccessLayer/ReKreator.DataAccessLayer.csproj [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    <Reference Include="EntityFramework.SqlServer, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089, processorArchitecture=MSIL">
+      <HintPath>..\packages\EntityFramework.6.2.0\lib\net45\EntityFramework.SqlServer.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.AspNet.Identity.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.AspNet.Identity.Core.2.2.1\lib\net45\Microsoft.AspNet.Identity.Core.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.AspNet.Identity.EntityFramework, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.AspNet.Identity.EntityFramework.2.2.1\lib\net45\Microsoft.AspNet.Identity.EntityFramework.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.AspNet.Identity.Owin, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (AlexKonon) [inline src/ReKreator.DataAccessLayer/ReKreator.DataAccessLayer.csproj]: Please clean up unnecessary references.
I suppose that only asp.net indentity ef reference will cover all identity needs for the DAL.

File: src/ReCreator.MailDelivery/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
```
Comment (AlexKonon) [inline src/ReCreator.MailDelivery/Properties/AssemblyInfo.cs]: Clean up all occurrences of the unnecessary references across the solution.

File: src/ReKreator.MVC/ReKreator.MVC.csproj [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+      <HintPath>..\packages\EntityFramework.6.2.0\lib\net45\EntityFramework.dll</HintPath>
+    </Reference>
+    <Reference Include="EntityFramework.SqlServer, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089, processorArchitecture=MSIL">
+      <HintPath>..\packages\EntityFramework.6.2.0\lib\net45\EntityFramework.SqlServer.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.AspNet.Identity.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.AspNet.Identity.Core.2.2.1\lib\net45\Microsoft.AspNet.Identity.Core.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.AspNet.Identity.EntityFramework, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.AspNet.Identity.EntityFramework.2.2.1\lib\net45\Microsoft.AspNet.Identity.EntityFramework.dll</HintPath>
```
Comment (AlexKonon) [inline src/ReKreator.MVC/ReKreator.MVC.csproj]: Not sure, but I think that ef reference for the identity is redundant here.

File: src/ReKreator.MVC/Global.asax.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (AlexKonon) [inline src/ReKreator.MVC/Global.asax.cs]: Clean up references.

File: src/ReCreator.MailDelivery/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("ReCreator.MailDelivery")]
```
[the remark this answers is not in the document]
Comment (xDAndrew) [reply src/ReCreator.MailDelivery/Properties/AssemblyInfo.cs]: Fixed

File: src/ReKreator.Scheduler/Program.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+﻿namespace ReKreator.Scheduler
+{
+    internal class Program
```
[the remark this answers is not in the document]
Comment (xDAndrew) [reply src/ReKreator.Scheduler/Program.cs]: Fixed

**Pull request merged**